### DISCUSSION TITLE
Hex Color Permission

### DISCUSSION THREAD

**Discussion by Plum1111** [Zrips/CMI#4337] [Feature Requests]
**Description of feature/improvement request:**
Basically just a permission for hex colors so that you can restrict who can use them and who can't.

---
### Detailed explanation of what it should do, when and in what situation: ###
If they have the permission they can use them and if they don't they can't. It would work like the current color code format (cmi.colors.[type].[colorname]) but in colorname it would simply be 'hex' like cmi.colors.privatemessage.hex

### Basic explanation as to why this feature is needed and/or useful to have: ###
Useful if you want a certain rank to only be able to use hex colors

**Comment by CloudeLecaw** [comment]
Negate all you dont want them and give the starpermission after cmi.colors.privatemeassage.*
On other groups just give the choosen color permissions you want.
For the other groups just give what permissions they are allowed to use. 
Dont forget that color permissions contain also the bold, italic, underlinednand and obscured (magic) permission.

**Comment by Plum1111** [comment]
> Negate all you dont want them and give the starpermission after cmi.colors.privatemeassage.*
> On other groups just give the choosen color permissions you want.
> For the other groups just give what permissions they are allowed to use.
> Dont forget that color permissions contain also the bold, italic, underlinednand and obscured (magic) permission.

This was already tried and it doesn't work.

**Comment by Royster-85** [comment]
This is something that should be added. As said, the format options: Bold, Italic, Magic etc are included in the same permission node for cmi.colors.[type].*

There's no way you can put a permission node to allow all 256 Million hex colors now available. Having a permission node like cmi.colors.[type].hex would allow the use of all the hex codes and prevent players using the format ones.
